test/LibraryManager.IntegrationTest/FileSaveRestoreTests.cs

```diff
@@ -0,0 +1,344 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using Microsoft.Test.Apex.VisualStudio;
+using Microsoft.Test.Apex.VisualStudio.Solution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Web.LibraryManager.IntegrationTest
+{
+    [TestClass]
+    public class FileSaveRestoreTests : VisualStudioLibmanHostTest
+    {
+        ProjectTestExtension _webProject;
+        ProjectItemTestExtension _libManConfig;
+        const string _projectName = @"TestProjectCore20";
+
+        [TestInitialize()]
+        public void initialize()
+        {
+            _webProject = Solution.ProjectsRecursive[_projectName];
+            _libManConfig = _webProject.Find(SolutionItemFind.FileName, "libman.json");
+        }
+
+        [TestCleanup]
+        public void Cleanup()
+        {
+            int count = Editor.Edit.UndoStack.Count;
+            Editor.Edit.Undo(count);
+
+            _libManConfig.Save();
+
+            // Delete restored libraries
+            string deletePath = Path.Combine(SolutionRootPath, _projectName, "wwwroot", "lib");
+            Directory.Delete(deletePath, true);
+        }
+
+        [TestMethod]
+        public void FileSaveRestore_Cdnjs_AddNewLibraryWithoutSpecifingFiles()
+        {
+            _libManConfig.Open();
+            string[] expectedFilesAndFolders = new[] {
+                "jquery.js",
+                "jquery.min.js",
+            };
+
+            Editor.Caret.MoveToExpression("\"version\": \"1.0\"");
+            Editor.Caret.MoveToEndOfLine();
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"defaultProvider\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "cdnjs", caseInsensitive: true);
+            Editor.KeyboardCommands.Type("cdnjs");
+
+            Editor.Caret.MoveToExpression("\"libraries\"");
+            Editor.Caret.MoveDown(2);
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "jquery", caseInsensitive: true, timeout: 5000);
```

Comment:
>5000 [](start = 104, length = 4)

This is a really long timeout.  Can we do something to make this shorter?

Reply:
get rid of this line as using new json file

---
In reply to: [194894203](https://github.com/aspnet/LibraryManager/pull/113#discussion_r194894203) [](ancestors = 194894203,194550054)